src/effects/DtmfGen.cpp

```diff
@@ -568,33 +578,41 @@ bool EffectDtmf::MakeDtmfTone(float *buffer, size_t len, float fs, wxChar tone,
 
 void EffectDtmf::UpdateUI(void)
 {
-   mDtmfDutyT->SetLabel(wxString::Format(wxT("%.1f %%"), dtmfDutyCycle));
+   auto &settings = mSettings;
+
+   mDtmfDutyT
+      ->SetLabel(wxString::Format(wxT("%.1f %%"), settings.dtmfDutyCycle));
    mDtmfDutyT->SetName(mDtmfDutyT->GetLabel()); // fix for bug 577 (NVDA/Narrator screen readers do not read static text in dialogs)
 
-   mDtmfSilenceT->SetLabel(wxString::Format(_("%.0f ms"), dtmfTone * 1000.0));
+   mDtmfSilenceT
+      ->SetLabel(wxString::Format(_("%.0f ms"), settings.dtmfTone * 1000.0));
    mDtmfSilenceT->SetName(mDtmfSilenceT->GetLabel()); // fix for bug 577 (NVDA/Narrator screen readers do not read static text in dialogs)
 
-   mDtmfToneT->SetLabel(wxString::Format(_("%.0f ms"), dtmfSilence * 1000.0));
+   mDtmfToneT
+      ->SetLabel(wxString::Format(_("%.0f ms"), settings.dtmfSilence * 1000.0));
    mDtmfToneT->SetName(mDtmfToneT->GetLabel()); // fix for bug 577 (NVDA/Narrator screen readers do not read static text in dialogs)
 }
 
 void EffectDtmf::OnSequence(wxCommandEvent & WXUNUSED(evt))
 {
-   dtmfSequence = mDtmfSequenceT->GetValue();
-   Recalculate();
+   auto &settings = mSettings;
+   settings.dtmfSequence = mDtmfSequenceT->GetValue();
+   settings.Recalculate(*this);
    UpdateUI();
 }
 
 void EffectDtmf::OnDuration(wxCommandEvent & WXUNUSED(evt))
 {
+   auto &settings = mSettings;
    SetDuration(mDtmfDurationT->GetValue());
-   Recalculate();
+   settings.Recalculate(*this);
```

Comment:
Can we not call this on mSettings directly?

Reply:
I anticipate that the means of communicating the settings into many of these functions might change might change in the future.  Binding a reference to the settings at the top will avoid the need to rewrite many lines again later.  I did that consistently, even in some short functions.